lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/core/expression/ExpressionScope.kt

```diff
@@ -865,200 +813,172 @@ public interface ExpressionScope {
+  public fun <U, V : ScalarValue<U>> min(vararg numbers: Expression<V>): Expression<V> =
+    callFn("min", *numbers)
 
   /** Returns the greatest of all given [numbers]. */
-  public fun max(vararg numbers: Expression<Number>): Expression<Number> = callFn("max", *numbers)
+  public fun <U, V : ScalarValue<U>> max(vararg numbers: Expression<V>): Expression<V> =
+    callFn("max", *numbers)
 
-  /** Returns the greatest of all given dp [numbers]. */
-  @JvmName("maxDp")
-  public fun max(vararg numbers: Expression<Dp>): Expression<Dp> = callFn("max", *numbers)
+  /** Returns the absolute value of [value], i.e. always a positive value. */
+  public fun <U, V : ScalarValue<U>> abs(value: Expression<V>): Expression<V> = callFn("abs", value)
 
   /**
    * Rounds [value] to the nearest integer. Halfway values are rounded away from zero.
    *
    * For example `round(const(-1.5))` evaluates to `-2`.
    */
-  public fun round(value: Expression<Number>): Expression<Number> = callFn("round", value)
-
-  /**
-   * Rounds [value] to the nearest integer. Halfway values are rounded away from zero.
-   *
-   * For example `round(const(-1.5.dp))` evaluates to `-2.dp`.
-   */
-  @JvmName("roundDp")
-  public fun round(value: Expression<Dp>): Expression<Dp> = callFn("round", value)
-
-  /** Returns the absolute value of [value], i.e. always a positive value. */
-  public fun abs(value: Expression<Number>): Expression<Number> = callFn("abs", value)
-
-  /** Returns the absolute dp value of [value], i.e. always a positive dp value. */
-  @JvmName("absDp") public fun abs(value: Expression<Dp>): Expression<Dp> = callFn("abs", value)
-
-  /** Returns the smallest integer that is greater than or equal to [value]. */
-  public fun ceil(value: Expression<Number>): Expression<Number> = callFn("ceil", value)
+  public fun round(value: Expression<FloatValue>): Expression<IntValue> = callFn("round", value)
 
   /** Returns the smallest integer that is greater than or equal to [value]. */
-  @JvmName("ceilDp") public fun ceil(value: Expression<Dp>): Expression<Dp> = callFn("ceil", value)
-
-  /** Returns the largest integer that is less than or equal to [value]. */
-  public fun floor(value: Expression<Number>): Expression<Number> = callFn("floor", value)
+  public fun ceil(value: Expression<FloatValue>): Expression<IntValue> = callFn("ceil", value)
 
   /** Returns the largest integer that is less than or equal to [value]. */
-  @JvmName("floorDp")
-  public fun floor(value: Expression<Dp>): Expression<Dp> = callFn("floor", value)
+  public fun floor(value: Expression<FloatValue>): Expression<IntValue> = callFn("floor", value)
 
   /** Returns the shortest distance in meters between the evaluated feature and [geometry]. */
-  public fun distance(geometry: Expression<Geometry>): Expression<Number> =
+  public fun distance(geometry: Expression<GeoJsonValue>): Expression<FloatValue> =
     callFn("distance", geometry)
```

Comment:
With the new typing, shouldn't there be a `public fun const(geoJson: GeoJson)`, then?

Reply:
We didn't have it with the old typing either, but yeah that's a missing piece rn